src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -339,11 +365,22 @@ uct_dc_mlx5_dci_pool_init_dci(uct_dc_mlx5_iface_t *iface, uint8_t pool_index,
     return UCS_OK;
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_ep_init_hw_dcs(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
+{
+    if (ucs_array_is_empty(&iface->tx.dcis)) {
+        uct_dc_mlx5_iface_resize_and_fill_dcis(iface, 1);
+        uct_dc_mlx5_dci_pool_init_dci(iface, uct_dc_mlx5_ep_pool_index(ep),
+                                      UCT_DC_MLX5_HW_DCI);
```

Comment:
does it mean HW DCS is the first DCI, that is most likely to be used?

Reply:
It doesn't mean that, because in alloc_or_create we are allocating by the length of dcis array:
![image](https://github.com/user-attachments/assets/4cc83400-f2c6-4d4b-9c74-43b09a5e682a)